pkg/manager/member/tikv_scaler.go

```diff
@@ -103,16 +103,30 @@ func (tsd *tikvScaler) ScaleIn(meta metav1.Object, oldSet *apps.StatefulSet, new
 		return fmt.Errorf("tikvScaler.ScaleIn: failed to get pods %s for cluster %s/%s, error: %s", podName, ns, tcName, err)
 	}
 
+	tc, _ := meta.(*v1alpha1.TidbCluster)
+	upNumber := 0
+	storeState := ""
+	for _, store := range tc.Status.TiKV.Stores {
+		if store.State == v1alpha1.TiKVStateUp {
+			upNumber++
+		}
+		if store.PodName == podName {
+			storeState = store.State
+		}
+	}
+	if upNumber < 3 {
+		return fmt.Errorf("the number of stores in Up state of TidbCluster [%s/%s] is %d, less than 3, can't scale in TiKV, podname %s ", meta.GetNamespace(), meta.GetName(), upNumber, podName)
```

Comment:
If we return an error here, the TC will be enqueued and synced immediately over and over again, however, the resync is useless, so we just print the log here and return nil, in this case, the TC will be resynced every 30s or until next update from the user.